### ISSUE TITLE
SQL Injection in DataManagement API

### ISSUE THREAD

**Issue by DominikPinsel** [eclipse-dataspaceconnector/DataSpaceConnector#1605]
# Bug Report

## Describe the Bug

It's possible to manipulate a call to the DataManagementAPI to execute any SQL query. In the test this was done using the `/agreements/` endpoint. But as all SQL extension share the same code to build SQL queries, other endpoints might be affected, too.

### Expected Behavior
- No SQL injection possible

### Observed Behavior
- SQL injection possible

## Steps to Reproduce
1. Setup Connector with DataManagement API and SQL Extensions
2. Create a table called _test_ in the database
3. Execute the following API call to delete the table

```bash
curl -X GET "$providerDataMgmtUrl/$dataMgmtPath/contractagreements?filter=policy.id'LIKE/**/?/**/LIMIT/**/?/**/OFFSET/**/?;DROP/**/TABLE/**/test/**/--%20=%20ABC--" --header "$apiKey: $apiKeyValue" --header "Content-Type: application/json"
```

## Context Information

It looks like queries that target objects, that are stored as JSON in the database, are affected.

## Detailed Description
-

## Possible Implementation
-

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by paullatzelsperger** [comment]
@ndr-brt Essentially, we could introspect the left-hand operand (e.g. `policy.id`) and see whether that represents a valid traversal through the object graph, i.e. whether the path exists or not. In this example the left-hand operand would be `policy.id'LIKE/**/?/**/L....` which is clearly invalid. I would call this a "validation of the canonical format". 

However, we'll also have to investigate, whether a similar situation can arouse from the right-hand operand. I doubt it, because we're using prepared statements, but still.

**Comment by ndr-brt** [comment]
Yes, it seems feasible, on the right part we don't need to bother as you said, because of prepared statements.

**Comment by paullatzelsperger** [comment]
with class-based introspection we can achieve _some_ level of validation, but we will hit roadblocks when dealing with a) polymorphic types (e.g. `Constraint`) or with `Map` types.

So for now we'll provide basic, class-based validation with the caveat that query expressions, that target polymorphic or Map types, are disallowed (i.e. return a 400).
